### ISSUE TITLE
PreferenceCategory use "iconSpaceReserved" not work

### ISSUE THREAD

**Issue by tcqq** [Gericop/Android-Support-Preference-V7-Fix#132]
I tried using "iconSpaceReserved" into "PreferenceCategory", but it did not work.

How to create "PreferenceCategory  with picture red box same effect?

![image](https://user-images.githubusercontent.com/10644057/33803388-cb2ab03c-ddc9-11e7-842d-8d076ad0eefd.png)

https://material.io/guidelines/patterns/settings.html#settings-usage

**Comment by Gericop** [comment]
Yeah, it wouldn't work because the categories has no icons. I'll see if the padding can be implemented though.

**Comment by TCQQ** [comment]
Ok.

**Comment by tcqq** [comment]
Hi, is this problem now supported?

**Comment by Gericop** [comment]
`PreferenceCategory` still does not support the `iconSpaceReserved` attribute properly. Reported the issue to Google: https://issuetracker.google.com/issues/111662669. This library will provide a temporary fix for the AndroidX version in the _unreserved-way_ (meaning that icon space reservation won't be forced on smaller screen devices, thus, only the `false` value should be used, if needed at all).

**Comment by Gericop** [comment]
The AndroidX version's v**1.0.0-alpha2** contains the temporary fix.

**Comment by gregkorossy** [comment]
The reported issue says it was fixed, however, the official lib v1.0.0 still doesn't contain the fixes, so this lib keeps providing the bug fix.

**Comment by yccheok** [comment]
@Gericop Thanks for providing the fix. Currently, our app doesn't use AndroidX. But, we do use Android Architecture Component in the following way.

    def lifecycle_version = "1.1.1"
    // ViewModel and LiveData
    implementation "android.arch.lifecycle:extensions:$lifecycle_version"
    // alternately - if using Java8, use the following instead of compiler
    implementation "android.arch.lifecycle:common-java8:$lifecycle_version"


    def room_version = "1.1.1"
    implementation "android.arch.persistence.room:runtime:$room_version"
    annotationProcessor "android.arch.persistence.room:compiler:$room_version"

    def work_version = "1.0.0-alpha09"
    implementation "android.arch.work:work-runtime:$work_version" // use -ktx for Kotlin
    // optional - Firebase JobDispatcher support
    implementation "android.arch.work:work-firebase:$work_version"

Your `implementation 'com.takisoft.fix:preference-v7:28.0.0.0'` works well for us except Google breaks `app:iconSpaceReserved="false"` in `PreferenceCategory`.

Is it suitable for us to use your `The AndroidX version's v1.0.0-alpha2` for our case, or can the temporary fix on `PreferenceCategory` can be carry forward to `v7:28.0.0.0`?

Thank you very much.

**Comment by alihasan95** [comment]
How remove the padding from PreferenceCategory

**Comment by zhanghai** [comment]
Check out [my answer on StackOverflow](https://stackoverflow.com/a/52960668) for overriding a `<dimen>` to fix this.
